### PR TITLE
Replace usages of ptr::offset with ptr::{add,sub}.

### PR DESCRIPTION
**Pull request by frewsxcv**
Rust provides these helper methods – so let's use them!

### CODE REVIEW

**Comment by frewsxcv** [comment]
Relevant clippy issue: https://github.com/rust-lang-nursery/rust-clippy/issues/3047

**Comment by withoutboats** [comment]
@r? @eddyb

File: src/libcore/slice/mod.rs [hunk 1]
```diff
@@ -1843,7 +1843,7 @@ impl<T> [T] {
             let mut_ptr = rest.as_mut_ptr();
             (left,
              from_raw_parts_mut(mut_ptr as *mut U, us_len),
-             from_raw_parts_mut(mut_ptr.offset((rest.len() - ts_len) as isize), ts_len))
+             from_raw_parts_mut(mut_ptr.add(rest.len()), ts_len))
```
[the remark this answers is not in the document]
Comment (frewsxcv) [reply src/libcore/slice/mod.rs]: Wow, very good catch 😓 

Addressed in the latest force push

[the remark this answers is not in the document]
Comment (RalfJung) [reply src/libcore/slice/mod.rs]: Dang, seems like I was not as thorough as I should have been :/  Thanks @mikerite!

@frewsxcv no need to force-push here I think, but whatever.

**Comment by RalfJung** [comment]
Bors timeout? @rust-lang/infra 

@bors retry

**Comment by kennytm** [comment]
The macs run for more than 4 hours on Travis.

**Comment by RalfJung** [comment]
What is this?
```
test jobserver::makes_jobserver_used ... FAILED
[...]
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 10054, kind: ConnectionReset, message: "An existing connection was forcibly closed by the remote host." }', libcore\result.rs:983:5
```
Looks like network trouble to me.

@bors retry

**Pull request merged**